src/js/network-vis.js

```diff
@@ -108,9 +117,47 @@ export function render(json) {
         //errors if there are duplicate items
         console.log('Encountered error: ' + error);
       }
+    } else if (element.GRANTED_ON_DATABASE) {
+      const index = hierarchy.findIndex((db) => db.text === element.GRANTED_ON_DATABASE);
+      database = index !== -1 ? hierarchy[index] : { text: element.GRANTED_ON_DATABASE, children: [] };
+
+      if (element.GRANTED_ON_SCHEMA) {
+        let schemaIndex = database.children.findIndex((schema) => schema.text === element.GRANTED_ON_SCHEMA);
+        schema =
+          schemaIndex !== -1 ? database.children[schemaIndex] : { text: element.GRANTED_ON_SCHEMA, children: [] };
+
+        if (element.GRANTED_ON_TYPE && element.GRANTED_ON_TYPE !== 'SCHEMA') {
+          let categoryIndex = schema.children.findIndex((category) => category.text === element.GRANTED_ON_TYPE);
+          category =
+            categoryIndex !== -1 ? schema.children[categoryIndex] : { text: element.GRANTED_ON_TYPE, children: [] };
+          categoryIndex === -1 ? schema.children.push(category) : (schema.children[categoryIndex] = category);
+
+          let itemIndex = category.children.findIndex((item) => item.text === element.GRANTED_ON_NAME);
+          item = itemIndex !== -1 ? category.children[itemIndex] : { text: element.GRANTED_ON_NAME, children: [] };
+          itemIndex === -1 ? category.children.push(item) : (category.children[itemIndex] = item);
+        }
+
+        schemaIndex === -1 ? database.children.push(schema) : (database.children[schemaIndex] = schema);
+      }
+      index === -1 ? hierarchy.push(database) : (hierarchy[index] = database);
+    } else {
+      let otherIndex = hierarchy.findIndex(
+        (other) => other.text === '(' + element.GRANTED_ON_TYPE + ')' + element.GRANTED_ON_NAME
+      );
+      other =
+        otherIndex !== -1
+          ? hierarchy[otherIndex]
+          : {
+              text: '(' + element.GRANTED_ON_TYPE + ')' + element.GRANTED_ON_NAME,
+              type: element.GRANTED_ON_TYPE,
+              children: [],
+            };
+
+      otherIndex === -1 ? hierarchy.push(other) : (hierarchy[otherIndex] = other);
     }
   });
-
+  hierarchy.sort((a, b) => (a.text > b.text ? -1 : 1));
```

Comment:
why sort?

Reply:
The folders will be jumbled up based on the data and hard to find the one you are looking for.